packages/labs/image/src/types.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * Checks if a string is a supported output image format.
+ * 
+ * @param value Value to be validated.
+ * @returns {Boolean} true if the string is a supported output format, false otherwise.
+ */
+export function isOutputFormat(value: string): value is OutputFormat {
+	return ['avif', 'jpeg', 'png', 'webp'].includes(value);
+}
+
+/**
+ * Checks if a string is a valid aspect ratio, ex: "16:9" or "3:4".
+ * 
+ * @param value Value to be validated.
+ * @returns {Boolean} true if the string is a valid aspect ratio, false otherwise.
+ */
+export function isAspectRatioString(value: string): value is `${number}:${number}` {
+	return /^\d*:\d*$/.test(value);
+}
+
+export type OutputFormat =
+	| 'avif'
+	| 'jpeg'
+	| 'png'
+	| 'webp';
+
+/**
+ * Converts a set of image props to the filename to use when building for static.
+ * 
+ * This is only used for static production builds and ignored when an SSR adapter is used,
+ * or in `astro dev` for static builds.
+ */
+export type FilenameFormatter = (props: ImageProps) => string;
+
+export interface IntegrationOptions {
+	/**
+	 * Entry point for the @type {HostedImageService} or @type {LocalImageService} to be used.
+	 */
+	serviceEntryPoint?: string;
+}
+
+/**
+ * Defines the original image and transforms that need to be applied to it.
+ */
+export interface ImageProps {
```

Comment:
This will be the basic interface used, third-party services can build on top to support more props like `blur`, `rotate`, etc